File: src/Niche/Prolog/prototype.pl
clone_profile(Profile, Clone) :-
    copy_term(Profile, Clone).

describe(profile(Name, Features), Text) :-
    atomic_list_concat(Features, ',', Joined),
    atomic_list_concat([Name, ': ', Joined], Text).

run :-
    Original = profile(orders, [metrics]),
    clone_profile(Original, profile(_, BaseFeatures)),
    append(BaseFeatures, [tracing], CanaryFeatures),
    Canary = profile('orders-canary', CanaryFeatures),
    describe(Original, OriginalText),
    describe(Canary, CanaryText),
    format('original=~w~n', [OriginalText]),
    format('clone=~w~n', [CanaryText]).

File: src/DataScience/R/prototype.R
clone_profile <- function(profile) {
  list(
    name = profile$name,
    features = as.character(profile$features)
  )
}

describe <- function(profile) {
  paste0(profile$name, ": ", paste(profile$features, collapse = ","))
}

original <- list(name = "orders", features = c("metrics"))
canary <- clone_profile(original)
canary$name <- "orders-canary"
canary$features <- c(canary$features, "tracing")

cat("original=", describe(original), "\n", sep = "")
cat("clone=", describe(canary), "\n", sep = "")
